lib/puppet/ssl/host.rb

```diff
@@ -419,6 +431,174 @@ def load_crls(path)
     end
   end
 
+  # Ensures that the CA certificate is available for either generating or
+  # validating the host's cert.
+  # It will first check if the cert is present in memory (used for testing),
+  # then check on disk, and finally try to download it.
+  # @raise [Puppet::Error] if text form of found certificate bundle is invalid
+  #                        and cannot be loaded into cert objects
+  # @return [Boolean] true if the CA certificate was found, false otherwise
+  def ensure_ca_certificate
+    file_path = certificate_location(CA_NAME)
+    if check_for_certificate_in_memory(CA_NAME)
+      true
+    elsif Puppet::FileSystem.exist?(file_path)
+      if load_certificate_bundle(Puppet::FileSystem.read(file_path))
+        true
+      else
+        raise Puppet::Error, _("The CA certificate at %{file_path} is invalid. Could not load.") % { file_path: file_path }
```

Comment:
why not let the error propagate from load_certificate_bundle here we we can use that information to be more specific about why it didn't load

Reply:
Justin had suggested doing it this way, see https://github.com/puppetlabs/puppet/pull/6824#discussion_r188053950. I don't have a strong opinion, given I wasn't using the information about why it didn't load (the error from OpenSSL is extremely unhelpful).

Reply:
Justin wasn't a fan of raising in one method only to immediately catch and transform to a different error, but I think in order to both preserve the specific error and provide context about where it came from (a problem on the server vs a problem on the local disk), we may need to do that.

Reply:
Do you think it's a good idea to print/log the text it tried and failed to load as a cert? We were going back and forth on that a couple days ago, but since you mentioned information...